pcg/src/main/kotlin/nl/perfana/config/generator/PerfanaConfigGeneratorController.kt

```diff
@@ -71,27 +78,41 @@ class PerfanaConfigGeneratorController(val storage: FileStorage) {
             }
         } finally {
             val output = extractOutput(process)
-            print(output)
+            println(output)
             process.destroy()
         }
     }
 
     private fun extractOutput(process: Process): String {
         val reader = BufferedReader(InputStreamReader(process.inputStream))
-        val output = reader.lines().collect(Collectors.joining(System.getProperty("line.separator")))
+        val output = reader.lines().collect(Collectors.joining(System.lineSeparator()))
         return output
     }
 
-    @GetMapping("/download/{projectId:.+}")
-    fun download(@PathVariable projectId: String, request: HttpServletRequest): ResponseEntity<Resource> {
+
+    @PostMapping("/generate")
+    fun generate(@RequestParam file: MultipartFile):  ResponseEntity<Resource> {
+        val timeStamp = DATE_TIME_FORMATTER.format(java.time.LocalDateTime.now())
+        val workDir = storage.createProjectDirectory(file, timeStamp)
+
+        // this should be done in a co routine
```

Comment:
co routine was to make it async processing, not need for this if we make it sync